### PR TITLE
feature/project-api-restoration

### PR DESCRIPTION
**Pull request by cdaringe**
#### asana
https://app.asana.com/0/45796090191289/137682875267750
# problem statement

- `core.projects` wasn't wired.
- `core.projects` kind of stunk.

# solution

- add projects api back in.
- make it sensible.
- test it.

### CODE REVIEW

File: src/model-service.js [hunk 1]
```diff
@@ -85,6 +85,30 @@ class ModelService {
...
+   * delete/remote a doc/model
+   * @param {object} model
+   * @param {function} [cb]
+   * @returns {Promise}
+   */
+  delete(model, cb) {
+    return nativepify(bluebird.resolve(
+      this.db.delete(model)
+    ).asCallback(cb));
+  }
```
Comment (cdaringe) [inline src/model-service.js]: FYI, we can drop `nativepify` when we bump to electron-prebuilt `1.2.0`

File: src/utils/file-stats.js [hunk 2]
```diff
@@ -5,54 +5,58 @@
  * Use the `projects` lib in order to access files.
  */
 'use strict';
```
Comment (cdaringe) [inline src/utils/file-stats.js]: i'd like to maybe build a `joi` schema for the filemeta that we cram into `project.files`. TBD, not now

File: src/model-service.js (hunk 1 above)
Comment (swashcap) [inline src/model-service.js]: :+1:

**Comment by swashcap** [comment]
:+1: Yep! Looks good.

**Pull request merged**